third_party_libraries/kml.js

```diff
@@ -501,99 +504,84 @@ L.Util.extend(L.KML, {
 
 	},
 
-
+	/**
+	 * Goes through a placemark kml element and generates a leaflet marker, based
+	 *		based on how many areas each placemark has
+	 *
+	 * @param {kml Element} place A placemark kml element
+	 * @param {xml} xml Xml data related to the placemark
+   * @param {path} style Style options for the placemark
+	 * @return {layer[]} layerArray An array of layers that contain leaflet
+	 * 		markers
+	 */
 	parsePlacemark: function (place, xml, style) {
-		var i, j, el, options = {};
-		el = place.getElementsByTagName('styleUrl');
-
-		for (i = 0; i < el.length; i++) {
-			var url = el[i].childNodes[0].nodeValue;
-			for (var a in style[url])
-			{
-				// for jshint
-				if (true)
-				{
-					options[a] = style[url][a];
-				}
+		var layerArray = [];
+		var areas = this.parseExtendedData(place);
+
+	 	for (area in areas) {
+			var options = this.getOptions(place, style)
+			var layer = this.initializeLayer(place, options, xml);
+			var name = this.parseName(place);
+			var description = this.parseDescription(place);
+
+			if (layer) {
+				layer.bindPopup("<h1>" + name + "</h1><b1>" + description + "</b1>", {offset: new L.Point(0,-20)});
```

Comment:
I'm really not a fan of creating HTML by appending tag strings together and it increases the possibility of XSS a lot of the time. Is there a better way to go about this? I know javascript has libraries for building HTML like this. I'd be more comfortable even if there was just a sanitization call for the 'name' and 'description' variables.